### PR TITLE
LPS-103180 Simplifies logic and splits in components for simplicity

### PR DESCRIPTION
**Pull request by jbalsas**
Hey @diegonvs! I've finally managed to review the `TranslationManager`. I've added some modifications... mostly because I'm bad with TOC, but also because I think we want to strive for simplicity as much as we can.

I've split the component in smaller pieces and reused Clay components a bit more when possible.

See some of the comments inline.

Could you please double check this and make changes if you spot issues before sending it to Julien for a final review?

Thanks!

### CODE REVIEW

File: modules/apps/data-engine/data-engine-taglib/src/main/resources/META-INF/resources/data_layout_builder/js/DataLayoutBuilder.es.js [hunk 1]
```diff
@@ -52,13 +57,8 @@ class DataLayoutBuilder extends Component {
...
 	disposed() {
-		if (this._translationManager) {
-			this._translationManager.detach(
-				'availableLocales',
-				this.onAvailableLocalesRemoved.bind(this)
-			);
-
-			this._translationManager = null;
+		if (this._translationManagerHandles) {
+			this._translationManagerHandles.forEach(handle => handle.detach());
```
Comment (jbalsas) [inline modules/apps/data-engine/data-engine-taglib/src/main/resources/META-INF/resources/data_layout_builder/js/DataLayoutBuilder.es.js:61]: The common pattern is that `.on` returns an event handle that can be detached

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/admin/js/main.es.js [hunk 2]
```diff
@@ -85,74 +85,81 @@ class Form extends Component {
...
-		Promise.all(dependencies).then(results => {
-			this._translationManager = results[2];
-
-			const translationManager = this._translationManager;
+		Promise.all(dependencies).then(
+			([
+				nameEditor,
+				descriptionEditor,
+				translationManager,
+				settingsDDMForm
```
Comment (jbalsas) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/admin/js/main.es.js:100]: Just destructured the array here for clarity, rather than using positional `results`... this is as fragile as before, but at least is slightly more readable

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/admin/js/main.es.js [hunk 3]
```diff
@@ -260,14 +267,17 @@ class Form extends Component {
 			this._autoSave.dispose();
 		}
 
+		if (this._stateSyncronizer) {
```
Comment (jbalsas) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/admin/js/main.es.js:270]: For some reason we weren't disposing this

File: modules/apps/frontend-taglib/frontend-taglib/build.gradle [hunk 4]
```diff
@@ -10,7 +10,6 @@ dependencies {
 	compileOnly project(":apps:frontend-taglib:frontend-taglib-clay")
 	compileOnly project(":apps:frontend-taglib:frontend-taglib-react")
 	compileOnly project(":apps:frontend-taglib:frontend-taglib-soy")
-	compileOnly project(":apps:portal-template:portal-template-react-renderer-api")
```
Comment (jbalsas) [inline modules/apps/frontend-taglib/frontend-taglib/build.gradle]: No longer needed, I think...

File: modules/apps/frontend-taglib/frontend-taglib/src/main/resources/META-INF/resources/translation_manager/hooks/useRegistry.js [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+		if (stateValue) {
+			return stateValue;
+		}
+	};
+
+	const on = (stateName, callback) => {
+		eventsRef.current.push({callback, stateName});
+
+		return {
+			detach: () => detach(stateName, callback)
```
Comment (jbalsas) [inline modules/apps/frontend-taglib/frontend-taglib/src/main/resources/META-INF/resources/translation_manager/hooks/useRegistry.js:47]: Return a handle to detach the events independently

**Comment by diegonvs** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: modules/apps/data-engine/data-engine-taglib/src/main/resources/META-INF/resources/data_layout_builder/js/DataLayoutBuilder.es.js (hunk 1 above)
Comment (diegonvs) [reply modules/apps/data-engine/data-engine-taglib/src/main/resources/META-INF/resources/data_layout_builder/js/DataLayoutBuilder.es.js:61]: Looks fine!

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/admin/js/main.es.js (hunk 2 above)
Comment (diegonvs) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/admin/js/main.es.js:100]: I totally agree, `results[2]` was very weird but when migrating I just left the code that hasn't been affect as is.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/admin/js/main.es.js (hunk 3 above)
Comment (diegonvs) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/admin/js/main.es.js:270]: Changing from `disposeInternal` override to `disposed` in favor to use Metal's lifecycle. Debugging it, the problem was solved in this case.

File: modules/apps/frontend-taglib/frontend-taglib/build.gradle (hunk 4 above)
Comment (diegonvs) [reply modules/apps/frontend-taglib/frontend-taglib/build.gradle]: Yep! Since we aren't using React's ComponentTag more.

File: modules/apps/data-engine/data-engine-taglib/src/main/resources/META-INF/resources/data_layout_builder/js/DataLayoutBuilder.es.js (hunk 1 above)
Comment (diegonvs) [reply modules/apps/data-engine/data-engine-taglib/src/main/resources/META-INF/resources/data_layout_builder/js/DataLayoutBuilder.es.js]: Makes sense.

**Comment by diegonvs** [comment]
Pull request submitted to https://github.com/julien/liferay-portal/pull/160. [See changes here.](https://github.com/diegonvs/liferay-portal/compare/dba646ffe01e~1...dba646ffe01e) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*
